### PR TITLE
Update reviewer_checklist.md

### PR DESCRIPTION
**Pull request by lucaferranti**
Changes

- change `paper.tex` to `paper.pdf` . Asking to check for the tex is incorrect, since that should actually **not** have the list of authors, since that is included from the `header.tex` which is automatically generated from the `paper.yml`. I believe checking these technicalities is more editor's job than reviewer's job, hence I suggest we simply ask reviewers to check the final paper has a list of authors with affiliations.

- Clarified statement of need section (this is FAQ I get on almost every review, iirc we discussed we simply want the statement of need to be clearly conveyed from the paper.

### CODE REVIEW

File: .buffy/templates/reviewer_checklist.md [hunk 1]
```diff
@@ -31,8 +31,8 @@
 
 ### Paper format
 
-- [ ] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
-- [ ] **A statement of need:** Does the paper have a section titled 'Statement of need' that clearly states what problems the software is designed to solve, who the target audience is, and its relation to other work?
+- [ ] **Authors:** Does the `paper.pdf` file include a list of authors with their affiliations?
+- [ ] **A statement of need:** Does the paper clearly state what problems it is designed to solve, who the target audience is, and its relation to other work?
```
Comment (luraess) [inline .buffy/templates/reviewer_checklist.md:35]: Maybe we miss that the software should address some challenges or problem. Also, we could try phrasing things to highlight it should "describe the implications and importance of the problem to the wider community" as well. I currently just struggle to find good wording...

Comment (lucaferranti) [reply .buffy/templates/reviewer_checklist.md:35]: a bit plain, but what about

```suggestion
- [ ] **A statement of need:** Does the paper clearly state what problems it is designed to solve, how that is relevant to the wider community, who the target audience is, and its relation to other work?
``

Comment (lucaferranti) [reply .buffy/templates/reviewer_checklist.md:35]: or maybe
```suggestion
- [ ] **A statement of need**: Does the paper clearly explain the problem it addresses, its importance to the broader community, the intended audience, and how it connects to existing work in the field?
```

Comment (luraess) [reply .buffy/templates/reviewer_checklist.md:35]: I like the second option

**Review by luraess — APPROVED** [review]
Looks good to me now

**Pull request merged**